hog.py

```diff
@@ -248,8 +251,17 @@ def bacon_strategy(score, opponent_score):
     >>> bacon_strategy(50, 70)
     0
     """
-    "*** YOUR CODE HERE ***"
-    return 5 # Replace this statement
+
+    if (opponent_score < 7) or (opponent_score == 0):
```

Comment:
Hard-coding "7" in here is considered a "magic number". In general, magic numbers are things to be avoided in a code base and should be left to configuration in something like an environment variable, config file, database, etc. Ask yourself, where did the number "7" come from? Is it a universal like the value of Pi or is there something in the context of the problem we are working that this number came from? In this case, "7" came from the number right before the current value of the BACON_MARGIN global variable that could change it's value in any other instance of hog, therefore we should be putting the global variable here rather than directly hard-coding "7". This goes for the following lines where this value is also used.